File: Question_Go/7_Reverse_Integer.go
func reverse(x int) int {
    sign := 0
    if x < 0 {
        sign = -1
        x = -x
    } else {
        sign = 1
    }

    ans := 0
    for x != 0 {
        num := x % 10
        x /= 10

        //overflow or not
        if ans > math.MaxInt32 / 10 {
            return 0
        }
        ans = ans * 10 + num
    }

    return ans * sign
}

File: Question_Java/7_Reverse_Integer.java
class Solution {
    public int reverse(int x) {
        int ans = 0;
        while (x != 0) {
            if (Math.abs(ans) > Integer.MAX_VALUE / 10)
                return 0;
            ans = ans * 10 + x % 10;
            x /= 10;
        }
        return ans;
    }
}
